pilot/pkg/xds/proxy_dependencies.go

```diff
@@ -35,6 +35,14 @@ func ConfigAffectsProxy(req *model.PushRequest, proxy *model.Proxy) bool {
 		return true
 	}
 
+	// When endpoints for headless service change, we should push if EnableHeadlessService is true
+	// so that we create updated listeners or when DNSCapture is enabled for proxy so that the NDS
+	// table is updated.
+	if req.Full && len(req.Reason) == 1 && req.Reason[0] == model.HeadlessEndpointUpdate &&
```

Comment:
This is not very right, we could merge two push requests. If so the reasons index can be random

Reply:
Ah Right. I missed that part.

Reply:
@hzxuzhonghu & @ramaraochavali  One more question, I noticed that only one request reason can be added in L#77,  so where should more than one requests can be merged? Thanks!